LoopKit/HealthStoreUnitCache.swift

```diff
@@ -29,6 +33,9 @@ public class HealthStoreUnitCache {
         userPreferencesChangeObserver = NotificationCenter.default.addObserver(forName: .HKUserPreferencesDidChange, object: healthStore, queue: nil, using: { [weak self] (_) in
             _ = self?.unitCache.mutate({ (cache) in
                 cache.removeAll()
+                DispatchQueue.global().async {
```

Comment:
As I commented in the `Loop` PR, couldn't we walk the cache and see specifically *what* has changed? If it isn't anything in the cache, then we don't care about it, so we can ignore it. (Well, at least I'm assuming that all preferred units in Loop/LoopKit go through the unit cache.)